medium/users/services.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth.password_validation import validate_password
+from django.core.validators import validate_email
+
+from medium.users.models import User
+
+
+class UserService:
+    def create_user(self, name: str, email: str, password: str) -> User:
+        user: User = User(name=name)
+        validate_email(email)
+        user.email = email
+
+        validate_password(password, user)
+        hashed_password: str = make_password(password)
+        user.password = hashed_password
+
+        user.save()
+        return user
+
+    def update_user(
```

Comment:
while updating using a `dict` is more elegant than checking for the parameters  
I also recommend using genric `model_update` written in `common/services.py` 
refer to:
https://github.com/HackSoftware/Django-Styleguide#naming-convention

Reply:
I think receiving the updated fields using a single `dict` will be better than separated parameters, but I think using `model_update` is over-engineering.

Reply:
@ahmedeltaweel Please, can you share with us your opinion about this

Reply:
if you saw the implementation it's pretty compact and straightforward instead of manually checking & setting every field.

Reply:
I already read the implementation and I have to points on it
- We can implement the point you mentioned above in the main service instead of using `model_update` service, so anyone in the future needs to add some functionalty or update the service will not need to read the two services first to be able to do what he needs.
- I would prefer that the user view contact the user service directly using the user ID instead of making it fetches the user object then passing it to the user service.